src/hooks/useSetFocus/__tests__/useSetFocus.jest.js

```diff
@@ -0,0 +1,89 @@
+import { act, cleanup, renderHook } from "@testing-library/react-hooks";
+import useSetFocus from "../index";
+
+describe("useSetFocus", () => {
+  let element;
+  const focusCallback = jest.fn();
+  const blurCallback = jest.fn();
+
+  describe("click", () => {
+    afterEach(() => {
+      element.remove();
+      cleanup();
+    });
+
+    it("default isFocused value is false", () => {
+      const { result } = renderHookForTest();
+      expect(result.current.isFocused).toBe(false);
+    });
+
+    it("should call focusCallback when focusing the element", () => {
+      const { result } = renderHookForTest();
+
+      act(() => {
+        element.focus();
+      });
+
+      expect(blurCallback.mock.calls.length).toBe(0);
+      expect(focusCallback.mock.calls.length).toBe(1);
+      expect(result.current.isFocused).toBe(true);
+    });
+
+    it("should call blurCallback when blur even happens to the element", () => {
+      const { result } = renderHookForTest();
+
+      act(() => {
+        element.focus();
+      });
+
+      act(() => {
+        element.blur();
+      });
+
+      expect(blurCallback.mock.calls.length).toBe(1);
+      expect(result.current.isFocused).toBe(false);
+    });
+
+    it("element should be focused after calling hook.focus()", () => {
+      const { result } = renderHookForTest();
+
+      act(() => {
+        result.current.focus();
+      });
+
+      expect(blurCallback.mock.calls.length).toBe(0);
```

Comment:
same here, if we test that when the element has native focus it will trigger the right callbacks and change isFocused return value in separate tests - there is no need to check it again.